jade/tests/test_centralauth.py

```diff
@@ -0,0 +1,249 @@
+                    'editcount': 0,
+                    'registration': '2014-08-17T23:22:33Z',
+                },
+                {
+                    'wiki': 'metawiki',
+                    'url': 'https://meta.wikimedia.org',
+                    'timestamp': '2014-08-17T23:22:31Z',
+                    'method': 'new',
+                    'editcount': 0,
+                    'registration': '2014-08-17T23:22:31Z',
+                },
+                {
+                    'wiki': 'nowikimedia',
+                    'url': 'https://no.wikimedia.org',
+                    'timestamp': '2017-12-13T09:53:07Z',
+                    'method': 'login',
+                    'editcount': 0,
+                    'registration': '2017-12-13T09:53:07Z',
+                }
+            ]
+        }
+    }
+}
+
+autopatrolled_metawiki_gui_response = {
+    "batchcomplete": "",
+    "query": {
+        "globaluserinfo": {
+            "home": "enwiki",
+            "id": 15622560,
+            "registration": "2012-12-21T22:37:22Z",
+            "name": "Awight (WMF)",
+            "groups": [
+                "oathauth-tester",
+                "wmf-researcher"
+            ],
+            "merged": [
+                {
+                    "wiki": "metawiki",
+                    "url": "https://meta.wikimedia.org",
+                    "timestamp": "2012-12-21T23:59:21Z",
+                    "method": "login",
+                    "editcount": 222,
+                    "registration": "2012-12-21T23:59:21Z",
+                    "groups": [
+                        "autopatrolled"
+                    ]
+                },
+                {
+                    "wiki": "ptwiki",
+                    "url": "https://pt.wikipedia.org",
+                    "timestamp": "2013-04-09T01:26:35Z",
+                    "method": "login",
+                    "editcount": 0,
+                    "registration": "2013-04-09T01:26:35Z"
+                },
+            ],
+            "unattached": []
+        }
+    }
+}
+
+autopatrolled_metawiki_lui_response = {
+    'batchcomplete': '',
+    'query': {
+        'users': [{
+            'userid': 2043420,
+            'name': 'Awight (WMF)',
+            'groups': ['autopatrolled', '*', 'user', 'autoconfirmed']
+        }]
+    }
+}
+
+no_autopatrolled_metawiki_lui_response = {
+    'batchcomplete': '',
+    'query': {
+        'users': [{
+            'userid': 2043420,
+            'name': 'Awight (WMF)',
+            'groups': ['*', 'user', 'autoconfirmed']
+        }]
+    }
+}
+
+
+def test_from_config():
+    ca = CentralAuth.from_config(test_config)
+    assert isinstance(ca, CentralAuth)
+
+
+# TODO: We can't test this using the current mocking strategy, because the
+# error comes from within .get
+# def test_get_globaluser_info_invalid_user_id():
+#    ca = CentralAuth.from_config(test_config)
+#    with pytest.raises(mwapi.errors.APIError):
+#        ca.get_globaluser_info(0)
```

Comment:
Not sure I understand this TODO.

Reply:
It turns out, requesting non-existent users actually causes an APIError in mwapi, rather than the "missing" sad path.  We probably need to handle this, and calling functions will be expecting differents types of jade.errors.UserPermissionError.  We should consider wrapping the APIError...  Also, the TODO is because I hadn't decided which code we should be checking.  Now I think we should be mocking mwapi.Session.get to just throw an error, but first let's decide how the CentralAuth class will handle it.